src/terraform/install.sh

```diff
@@ -366,6 +387,49 @@ install_terraform() {
     curl -sSL -o ${terraform_filename} "${HASHICORP_RELEASES_URL}/terraform/${TERRAFORM_VERSION}/${terraform_filename}"
 }
 
+verify_signature() {
+    local gpg_key=$1
+    local sha256sums_url=$2
+    local sig_url=$3
+    local sha256sums_file=$4
+    local sig_file=$5
+
+    receive_gpg_keys "$gpg_key"
+    curl -sSL -o "$sha256sums_file" "$sha256sums_url"
+    curl -sSL -o "$sig_file" "$sig_url"
+    gpg --verify "$sig_file" "$sha256sums_file"
+}
```

Comment:
Can you merge `verify_sigature` and `verify_with_fallback` and the fallback fn is an optional param at the end so that, if it fails _and_ there if a fallback fn, it calls it? That way we avoid having two fns that do virtually the same thing